### ISSUE TITLE
string.Format gives enum number(s) not names

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#1312]
### Expected

A B C
A B
0 3 4
0 3
### Actual

0 3 4
0 3
0 3 4
0 3
### Steps To Reproduce

[Bridge Fiddle](http://live.bridge.net/#c9f9eb7fa3ab3d29a9be5c90fccbb24e)

``` csharp
public static class Program {
    public enum SimpleEnum {
        A,
        B = 3,
        C,
        D = 10
    }

    public enum ByteEnum : byte {
        A,
        B = 3,
    }
[Ready]
    public static void Main (string[] args) {
        Console.WriteLine(string.Format("{0} {1} {2}", SimpleEnum.A, SimpleEnum.B, SimpleEnum.C));
        Console.WriteLine(string.Format("{0} {1}", ByteEnum.A, ByteEnum.B));
        Console.WriteLine(string.Format("{0} {1} {2}", (int)SimpleEnum.A, (int)SimpleEnum.B, (int)SimpleEnum.C));
        Console.WriteLine(string.Format("{0} {1}", (int)ByteEnum.A, (int)ByteEnum.B));
    }
}
```

**Comment by geoffreymcgill** [comment]
Thanks for the excellent sample. We are reviewing.

**Comment by geoffreymcgill** [comment]
Related to #1290, #1292 and other type related issues that will need custom client side wrappers.

Cannot fix for Bridge 1.14 release.
